pkg/cri/resource-manager/policy/builtin/memtier/pools.go

```diff
@@ -774,10 +774,10 @@ func (p *policy) compareScores(request Request, pools []Node, scores map[int]Sco
 
 	// 1) a node with insufficient isolated or shared capacity loses
 	switch {
-	case (isolated2 < 0 && isolated1 >= 0) || (shared2 <= 0 && shared1 > 0):
+	case (isolated2 < 0 && isolated1 >= 0) || (shared2 < 0 && shared1 >= 0):
```

Comment:
Hmm... these score comparison changes together with the subsequent change in `Allocate()` basically revert 6d7ac67e3c211faf41c5a8a0a3b1a2e4cf3db3ed, which was put in place to explicitly disallow fully exhausting
the shared subset of a pool. I wonder if that really was/is an unnecessarily strict restriction (because TBH, I
can't recall the exact details right now when/how that caused breakage).

Reply:
Yeah, now I recall we've discussed this earlier. It seems back then (https://github.com/intel/cri-resource-manager/commit/bf1642bdfac8a0de09fa0480a0202b1f723bea2e) you mentioned a pending commit that would change this behavior. But I don't remember anymore what unwanted behavior it would unleash...

Reply:
That pending commit was dropped as it would have caused problems for containers in the BestEffort QoS class. Those come with a CPU request/limit of 0 milli-CPUs (which translates to 2 CFS cpus shares). These containers still need to be able run on some set of  CPUs. This set is the shared subset of the pool. If we allowed a pool to ever run out of shared CPUs and the pool had BestEffort QoS class containers assigned, then we'd end up trying to set cpuset.cpus for these containers to the empty set, which is then rejected by the kernel.

One easy kludge-around for this would be to account for BestEffort containers as having a request of  1 milli-CPU instead of 0. That would still leave the (theoretical) problem of Guaranteed QoS class containers first exhausting all the CPUs and then a scheduler BestEffort QoS class one getting assigned to the empty cpuset and failing.

IMO the real fix would be to really implement reserved CPUs and then co-locate BestEffort QoS class containers together with the ones in the kube-system namespace there.